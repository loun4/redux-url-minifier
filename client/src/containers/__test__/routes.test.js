
import React from 'react';
import renderer from 'react-test-renderer';
import RouterService from '../../utils/router-service';
import Routes from '../routes';

const buildSubject = () =>
  <Routes {...RouterService.state} />;

test('Render app', () => {
  const component = renderer.create(buildSubject());
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('/ render main component', () => {
  RouterService.navigate('/');
  const component = renderer.create(buildSubject());
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('/admin render main component', () => {
  RouterService.navigate('/admin');
  const component = renderer.create(buildSubject());
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
