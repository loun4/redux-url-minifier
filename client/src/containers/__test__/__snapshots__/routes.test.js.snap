// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/ render main component 1`] = `
<div>
  <div
    className="ui inverted menu"
  >
    <a
      className="header item active"
      href="/"
      onClick={[Function]}
    >
      URL shortener
    </a>
    <div
      className="right menu"
    >
      <a
        className="item"
        href="/admin"
        onClick={[Function]}
      >
        Admin
      </a>
    </div>
  </div>
  <div
    className="app ui container"
  >
    <div>
      <form
        className="ui form column massive separator"
        noValidate={undefined}
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <div
            className="ui action input"
          >
            <input
              autoComplete="off"
              className="field-input"
              disabled={false}
              id="link"
              name="linkForm.linkURL"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Past your original url"
              type="text"
              value=""
            />
            <button
              className="ui button"
              type="submit"
            >
              Go !
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      className="ui negative message notification"
    >
      Something worng happened, please try again.
    </div>
  </div>
</div>
`;

exports[`/admin render main component 1`] = `
<div>
  <div
    className="ui inverted menu"
  >
    <a
      className="header item"
      href="/"
      onClick={[Function]}
    >
      URL shortener
    </a>
    <div
      className="right menu"
    >
      <a
        className="item active"
        href="/admin"
        onClick={[Function]}
      >
        Admin
      </a>
    </div>
  </div>
  <div
    className="app ui container"
  >
    <div
      className="signin"
    >
      <div
        className="wide column"
      >
        <h2
          className="ui teal image header"
        >
          <div
            className="content"
          >
            Log-in
          </div>
        </h2>
        <form
          className="ui large form"
          noValidate={undefined}
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="ui stacked segment"
          >
            <div
              className="field"
            >
              <div
                className="ui left icon input"
              >
                <i
                  className="user icon"
                />
                <input
                  className="field-input"
                  disabled={false}
                  id="login"
                  name="signinForm.login"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyPress={[Function]}
                  placeholder="Login"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="field"
            >
              <div
                className="ui left icon input"
              >
                <i
                  className="lock icon"
                />
                <input
                  className="field-input"
                  disabled={false}
                  id="password"
                  name="signinForm.password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyPress={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
            </div>
            <button
              className="ui fluid large teal submit button"
              type="submit"
            >
              Login
            </button>
          </div>
          <div
            className=""
          >
            <ul
              className="list"
            />
          </div>
        </form>
      </div>
    </div>
    <div
      className="ui negative message notification"
    >
      Something worng happened, please try again.
    </div>
  </div>
</div>
`;

exports[`Render app 1`] = `
<div>
  <div
    className="ui inverted menu"
  >
    <a
      className="header item active"
      href="/"
      onClick={[Function]}
    >
      URL shortener
    </a>
    <div
      className="right menu"
    >
      <a
        className="item"
        href="/admin"
        onClick={[Function]}
      >
        Admin
      </a>
    </div>
  </div>
  <div
    className="app ui container"
  >
    <div>
      <form
        className="ui form column massive separator"
        noValidate={undefined}
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <div
            className="ui action input"
          >
            <input
              autoComplete="off"
              className="field-input"
              disabled={false}
              id="link"
              name="linkForm.linkURL"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Past your original url"
              type="text"
              value=""
            />
            <button
              className="ui button"
              type="submit"
            >
              Go !
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      className="ui negative message notification"
    >
      Something worng happened, please try again.
    </div>
  </div>
</div>
`;
